tests/transformers/HighlightedValueFactory.ts

```diff
@@ -86,4 +86,125 @@ it('Nested highlighted fields', () => {
   const actualHighlightedValues = HighlightedValueFactory.create(liveApiData);
 
   expect(actualHighlightedValues).toMatchObject(expectedHighlightedValues);
-});
\ No newline at end of file
+});
+
+it('Highlighted fields with arrays', () => {
+  const liveApiData = {
+    languages: [
+      {
+        value: 'English',
+        matchedSubstrings: []
+      },
+      {
+        value: 'Spanish',
+        matchedSubstrings: [
+          {
+            offset: 0,
+            length: 7
+          }
+        ]
+      }
+    ]
+  };
+
+  const expectedHighlightedValues = [
+    {
+      fieldName: 'languages',
+      value: 'English',
+      path: ['languages'],
+      matchedSubstrings: []
+    },
+    {
+      fieldName: 'languages',
+      value: 'Spanish',
+      path: ['languages'],
+      matchedSubstrings: [
+        {
+          offset: 0,
+          length: 7
+        }
+      ]
+    }
+  ];
+
+  const actualHighlightedValues = HighlightedValueFactory.create(liveApiData);
+  expect(actualHighlightedValues).toMatchObject(expectedHighlightedValues);
+});
+
+it('Highlighted fields with arbitrarily placed arrays', () => {
+  const liveApiData = {
```

Comment:
Can you add a nested field inside one of array items?

Reply:
yeah - nested arrays was technically being handled by the implementation, but I thought it wasn't possible so didn't add a test case for it. After talking with Nathan clearly it is possible!